underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/EntityGroupFilterBuilderBase.java

```diff
@@ -45,21 +45,80 @@ public EntityGroupFilterBuilderBase(CriteriaSelector criteriaSelector) {
     super(criteriaSelector);
   }
 
+  // A SelectionGroup is the parameters that must be the same to be applied to a list of ids.
+  protected class SelectionGroup implements Comparable<SelectionGroup> {
+    public String entityGroupId;
+    public ValueDataOuterClass.ValueData valueData;
+
+    public SelectionGroup(String entityGroupId, ValueDataOuterClass.ValueData valueData) {
+      this.entityGroupId = entityGroupId;
+      this.valueData = valueData;
+    }
+
+    public String getEntityGroupId() {
+      return entityGroupId;
+    }
+
+    @Override
+    public int compareTo(SelectionGroup sg) {
+      // A simple comparison for test stability.
```

Comment:
minor: check and return 1 if sg is null. if sg == this return 0.
reason: arraylist can hold null. if we sort / compare values in the list via a stream/other, this is needed